Compiled from "AsyncRetryingExecutor.java"
class com.amazonaws.services.simpleworkflow.flow.interceptors.AsyncRetryingExecutor$2 extends com.amazonaws.services.simpleworkflow.flow.core.TryCatchFinally {
  java.lang.Throwable failureToRetry;

  final com.amazonaws.services.simpleworkflow.flow.interceptors.AsyncRetryingExecutor this$0;

  private final com.amazonaws.services.simpleworkflow.flow.interceptors.AsyncRunnable val$command;

  private final com.amazonaws.services.simpleworkflow.flow.core.Settable val$shouldRetry;

  com.amazonaws.services.simpleworkflow.flow.interceptors.AsyncRetryingExecutor$2(com.amazonaws.services.simpleworkflow.flow.interceptors.AsyncRetryingExecutor, com.amazonaws.services.simpleworkflow.flow.interceptors.AsyncRunnable, com.amazonaws.services.simpleworkflow.flow.core.Settable);
    Code:
       0: aload_0
       1: aload_1
       2: putfield      #16                 // Field this$0:Lcom/amazonaws/services/simpleworkflow/flow/interceptors/AsyncRetryingExecutor;
       5: aload_0
       6: aload_2
       7: putfield      #18                 // Field val$command:Lcom/amazonaws/services/simpleworkflow/flow/interceptors/AsyncRunnable;
      10: aload_0
      11: aload_3
      12: putfield      #20                 // Field val$shouldRetry:Lcom/amazonaws/services/simpleworkflow/flow/core/Settable;
      15: aload_0
      16: invokespecial #22                 // Method com/amazonaws/services/simpleworkflow/flow/core/TryCatchFinally."<init>":()V
      19: aload_0
      20: aconst_null
      21: putfield      #25                 // Field failureToRetry:Ljava/lang/Throwable;
      24: return

  protected void doTry() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #18                 // Field val$command:Lcom/amazonaws/services/simpleworkflow/flow/interceptors/AsyncRunnable;
       4: invokeinterface #36,  1           // InterfaceMethod com/amazonaws/services/simpleworkflow/flow/interceptors/AsyncRunnable.run:()V
       9: return

  protected void doCatch(java.lang.Throwable) throws java.lang.Throwable;
    Code:
       0: aload_1
       1: instanceof    #43                 // class java/util/concurrent/CancellationException
       4: ifeq          9
       7: aload_1
       8: athrow
       9: aload_0
      10: aload_1
      11: putfield      #25                 // Field failureToRetry:Ljava/lang/Throwable;
      14: return

  protected void doFinally() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #20                 // Field val$shouldRetry:Lcom/amazonaws/services/simpleworkflow/flow/core/Settable;
       4: aload_0
       5: getfield      #25                 // Field failureToRetry:Ljava/lang/Throwable;
       8: invokevirtual #48                 // Method com/amazonaws/services/simpleworkflow/flow/core/Settable.set:(Ljava/lang/Object;)V
      11: return
}
